pinot-common/src/main/java/org/apache/pinot/common/protocols/SegmentCompletionProtocol.java

```diff
@@ -190,10 +204,10 @@ public String getUrl(String hostPort, String protocol) {
           + (_params.getSegmentSizeBytes() <= 0 ? ""
           : ("&" + PARAM_SEGMENT_SIZE_BYTES + "=" + _params.getSegmentSizeBytes())) + (_params.getNumRows() <= 0 ? ""
           : ("&" + PARAM_ROW_COUNT + "=" + _params.getNumRows())) + (_params.getSegmentLocation() == null ? ""
-          : ("&" + PARAM_SEGMENT_LOCATION + "=" + _params.getSegmentLocation()))
-          + (_params.getStreamPartitionMsgOffset() == null ? ""
-          : ("&" + PARAM_STREAM_PARTITION_MSG_OFFSET + "=" + _params.getStreamPartitionMsgOffset()))
-          ;
+          : ("&" + PARAM_SEGMENT_LOCATION + "=" + _params.getSegmentLocation())) + (
```

Comment:
Can we improve this by using some URI builder? I understand that we have been building URL in this way before this PR but it would be great to clean up since we are touching this part.

It will be cleaner if we keep adding parameters to `List<NameValuePair>` and use `new URI(...)` and then convert back to String?